File: Java/Best_Time_to_Buy_and_Sell_Stock.java
class Solution {
  public int maxProfit(int[] prices) {
    int min_price = Integer.MAX_VALUE;
    int profit = 0;
    for (int i : prices) {
      min_price = Math.min(min_price, i);
      profit = Math.max(profit, i - min_price);
    }
    return profit;
  }
}

File: Kotlin/Best_Time_to_Buy_and_Sell_Stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var minPrice = Int.MAX_VALUE
        var maxProfit = 0
        
        for (price in prices) {
            if (price < minPrice) {
                minPrice = price
            } else {
                val profit = price - minPrice
                maxProfit = maxOf(maxProfit, profit)
            }
        }
        return maxProfit
    }
}
